Frontend/src/Components/Dashboard/Dashboard.js

```diff
@@ -126,6 +136,9 @@ export default function Dashboard(props) {
         .then((res) => {
           //console.time("update react");
 
+          // when the car/data generator is offline, res.response is going to be null,
+          //  so we need to set variables to a default value that will prevent a TypeError
+          //  when we try reading the properties of an undefined variable
           setState({ data: res.response });

```

Comment:
```suggestion
          // when the car/data generator is offline, res.response is going to be null
          setState({ data: res.response });
```
@jackwherry, I apologize. One last nit: Since we aren't setting anything to a default value, the last two lines of comments can be removed.